docs/DeveloperGuide.md

```diff
@@ -2,36 +2,118 @@
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}  
+Zhou Zhou's Individual Project: https://github.com/maanyos/ip \
+Nay Chi's Individual Project: https://github.com/NayChi-7/ip
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Class Diagrams
+
+Biometrics component:  
+The Biometrics component stores information about the user's biometrics and holds a reference
+to an ArrayList of the user's weight and fat records under WeightAndFatList class.  
+WeightAndFatList is separated from Biometrics for better abstraction and cohesion.  
+[Proposed] Weight and fat records will be associated with a date, and the WeightAndFatList class
+stores records in descending order of date for viewing purposes.
+![Biometrics.png](UmlDiagrams/Biometrics.png)
+
+Food component:
+
+Food and FoodList are two classes implemented for storing user's food consumption.
+The interaction between the Food, FoodList class and Command classes(Addcommand, RemoveCommand,
+FindCommand and ViewCommand). Only food related details(i.e. methods and attributes) are added in the
+diagram below for readability.  
+![Food.png](UmlDiagrams/Food.png)
+
+\
+[Proposed] Food records will be associated with a date, and the FoodList class
+stores records in descending order of date for viewing purposes.
+
+CardioExercise/StrengthExercise and ExerciseList components:
+
+CardioExercise/StrengthExercise and ExerciseList are classes to store user's cardio/strength training exercises.
+Both CardioExercise and StrengthExercise inherit from Exercise and work similarly.
+CardioExercise/StrengthExcises are added, removed, found , mark done, mark undone and viewed from the ExerciseList with
+the use
+of AddCommand, RemoveCommand, FindCommand, MarkCommand and ViewCommand.\
+ExerciseList stores Exercise objects in two Arraylist based on the status of the exercise to save the time needed to
+filter exercise based on its status when user want to view the exercises based on its status.
+
+![](UmlDiagrams/StrengthExercise.png)
+
+[Proposed] StrengthExercise will be associated with a date, and the exercises will be displayed in descending order of
+date for viewing purposes.
+
+### Sequence diagrams
+
+Adding a new record  
+![Add.png](UmlDiagrams/Add.png)
+
+Adding weight and fat record
+![AddWeightAndFat.png](UmlDiagrams/AddWeightAndFat.png)
+
+Adding food record
+![AddFood.png](UmlDiagrams/AddFood.png)
+
+Adding strength exercise
+![](UmlDiagrams/AddStrengthExercise.png)
+
+Marking exercise
+![](UmlDiagrams/MarkExercise.png)
+
+Viewing historical records
+![View.png](UmlDiagrams/View.png)
+
+Viewing all historical records sorted by date
+![ViewAll.png](UmlDiagrams/ViewAll.png)
```

Comment:
In your view all diagram, there is alot of self calls to eventually get the UI output of the historical records. Perhaps you can consider hiding some of the calls? This is because there is alot of details that reader does not really have to know and it might confuse the user because after all what these calls does is to just to retrieve historical records and format it to output to the command line.